packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -114,7 +115,16 @@ class ConfigChainBuilder {
       options = jsonCache[content] = jsonCache[content] || json.parse(content);
       if (key) options = options[key];
     } catch (err) {
-      err.message = `${loc}: Error while parsing JSON - ${err.message}`;
+      const code = codeFrame(content, err.lineNumber, err.columnNumber);
+
+      err.message = `
+${loc}:
+${code}
+Error while parsing JSON - ${err.message}
+      `;
+
+      err.stack = err.stack.replace(/^\s*at\s.*:\d+:\d+[\s\)]*\n/gm, "");
```

Comment:
Personally I think I'd rather leave the stack trace as-is, since it can still provide useful information for debugging.